### DISCUSSION TITLE
How to access VIP functionality for worker nodes in hcloud?

### DISCUSSION THREAD

**Discussion by mcflis** [siderolabs/talos#7016] [Q&A] [answered]
Hey there,

we run our servers in Hetzner cloud and for our new k8s cluster, we are evaluating talos linux.

The built-in VIP feature is very helpful for control plane failover because it re-assigns the IP alias to a "ready" control plane node.

Now, we wanted to use https://github.com/metallb/metallb on worker nodes to expose a private load balancer ip in the 10.0.0.0/24 subnet. However, this won't work unless we assign the IP alias manually to a worker node in the Hetzner cloud console (or using `hcloud` cli).

**Since the update of IP aliases is already built into the hcloud platform image of talos for control plane nodes, we wanted to know if there is any way to make it available to talos worker nodes (either through machine config or a configurable container image)?** 

More info:

In our old cluster, we used pfsense and its built-in haproxy, but we'd rather like to handle things from within the new cluster instead of dealing with manual/decentralised configuration.

Also there is https://github.com/cbeneke/hcloud-fip-controller but it only works with floating ip resources an public ips

**Comment by smira** [comment, marked as answer]
Talos VIP feature is only for controlplane nodes, and only for Kubernetes controlplane endpoint.

Any other VIPs can be solved by running something on top of Kubernetes, it doesn't have to be built into Talos itself.

**Reply by mcflis** [reply]
Thanks for your reply.

I see, but what I meant is the specific hcloud api controlling code which uses the api token provided by https://www.talos.dev/v1.3/reference/configuration/#viphcloudconfig

It would be helpful to have this part of the talos linux image for hcloud available as separate container image so that anyone running in hcloud can take advantage of code that already exists to control IP aliases in hcloud. What do you think?

**Reply by smira** [reply]
you can extract that code and make it a controller which runs in Kubernetes. 

I'm not a HCloud user, so I don't know any specifics there.

My point was that this feature won't be part of the Talos Linux itself.
